src/anomalib/utils/callbacks/metrics_manager.py

```diff
@@ -0,0 +1,173 @@
+"""MetricsManager callback."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+from typing import Any
+
+import torch
+from lightning.pytorch import Callback
+from lightning.pytorch.utilities.types import STEP_OUTPUT
+from torch import Tensor
+
+from anomalib import trainer
+from anomalib.data import TaskType
+from anomalib.models import AnomalyModule
+from anomalib.utils.metrics import AnomalibMetricCollection, create_metric_collection
+
+logger = logging.getLogger(__name__)
+
+
+class _MetricsManagerCallback(Callback):
```

Comment:
Also, I've checked the other callbacks that are used by Trainer. They don't seem to be using `_Manager` suffix. To be more consistent, we could maybe rename this to `MetricsCallback`, or vice-versa?